crates/db_views/src/post/post_view.rs

```diff
@@ -32,36 +28,53 @@ use diesel::{
   TextExpressionMethods,
 };
 use diesel_async::RunQueryDsl;
+use i_love_jesus::asc_if;
 use lemmy_db_schema::{
   impls::local_user::LocalUserOptionHelper,
-  newtypes::{CommunityId, InstanceId, PersonId, PostId},
+  newtypes::{CommunityId, InstanceId, PaginationCursor, PersonId, PostId},
   source::{
-    keyword_block::LocalUserKeywordBlock,
     local_user::LocalUser,
-    post::{post_actions_keys, post_keys as key, Post, PostActionsCursor},
+    post::{post_keys as key, Post},
     site::Site,
   },
-  traits::Crud,
-  utils::{
-    fuzzy_search,
-    get_conn,
-    limit_and_offset,
-    now,
-    paginate,
-    seconds_to_pg_interval,
-    Commented,
-    DbPool,
-    ReverseTimestampKey,
-  },
+  traits::{Crud, PaginationCursorBuilder},
+  utils::{get_conn, limit_fetch, now, paginate, seconds_to_pg_interval, Commented, DbPool},
 };
 use lemmy_db_schema_file::{
-  enums::{CommunityFollowerState, CommunityVisibility, ListingType, PostSortType},
+  enums::{
+    CommunityFollowerState,
+    CommunityVisibility,
+    ListingType,
+    PostSortType::{self, *},
+  },
   schema::{community, community_actions, local_user_language, person, post, post_actions},
 };
-use tracing::debug;
-use PostSortType::*;
+use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+
+impl PaginationCursorBuilder for PostView {
+  type CursorData = Post;
+  fn to_cursor(&self) -> PaginationCursor {
+    PaginationCursor::new_single('P', self.post.id.0)
+  }
+
+  async fn from_cursor(
+    cursor: &PaginationCursor,
+    pool: &mut DbPool<'_>,
+  ) -> LemmyResult<Self::CursorData> {
+    let pids = cursor.prefixes_and_ids();
+    let (_, id) = pids
+      .as_slice()
+      .first()
+      .ok_or(LemmyErrorType::CouldntParsePaginationToken)?;
```

Comment:
This logic is used various times, how about adding a helper method on PaginationCursor?